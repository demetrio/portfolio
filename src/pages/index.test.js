import React from 'react';
import { render } from '@testing-library/react';
import Index from './index';
import * as Gatsby from 'gatsby';

beforeEach(() => {
	const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
	useStaticQuery.mockImplementation(() => ({
		site: {
			siteMetadata: {
				title: `Demetrio Alvarez`,
				description: `Demetrio Alvarez - Front End Developer`,
				siteUrl: 'https://demetrio.dev',
				author: `Demetrio Alvarez`,
				keywords: ['Demetrio', 'Alvarez', 'Portfolio', 'Developer', 'FrontEnd'],
			},
		},
	}));
});

describe('Index', () => {
	it('renders correctly', () => {
		const { container } = render(<Index />);
		expect(container).toMatchSnapshot();
	});
});
