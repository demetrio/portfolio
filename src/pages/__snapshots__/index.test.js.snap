// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<div>
  <div
    class="css-14e7r5m"
  >
    <header
      class="css-hnjfqn"
    >
      <a
        class="title css-1dtxhj1"
        font-weight="bold"
        href="/"
      >
        Demetrio Alvarez
      </a>
      <a
        class="css-1ed1fe6"
        href="/"
      >
        <img
          alt="avatar"
          class="logo"
          src="test-file-stub"
        />
      </a>
      <nav
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <a
          class="css-6uh4pm"
          href="/"
        >
          Home
        </a>
        <a
          class="css-6uh4pm"
          href="/about/"
        >
          About
        </a>
        <a
          class="css-6uh4pm"
          href="/projects/"
        >
          Portfolio
        </a>
      </nav>
    </header>
    <div
      class="css-1kdo87l"
    >
      <main>
        <h1
          class="css-1utp280"
        >
          Hi! I'm Demetrio and I'm a Front End Developer. Here you can find out more 
          <a
            href="/about/"
          >
            about me
          </a>
          , send me
           
          <a
            href="mailto:contact@demetrio.dev"
          >
            an email
          </a>
          , see my 
          <a
            href="/projects/"
          >
            portfolio
          </a>
           and my code at
           
          <a
            href="https://github.demetrio.dev"
          >
            Github
          </a>
           or add me to 
          <a
            href="https://linkedin.demetrio.dev"
          >
            LinkedIn
          </a>
          .
        </h1>
      </main>
    </div>
  </div>
</div>
`;
